sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -479,15 +432,16 @@ object DataWritingSparkTask extends Logging {
   }
 }
 
-private[v2] trait AtomicTableWriteExec extends V2TableWriteExec with SupportsV1Write {
+private[v2] trait TableWriteExec extends V2TableWriteExec with SupportsV1Write {
```

Comment:
Do we have classes that extend `V2TableWriteExec` but not `TableWriteExec`?

Reply:
yes we do, e.g., `WriteToDataSourceV2Exec`.

Reply:
Seems like the name should be `TableWriteExecWithV1Support`? It's hard to tell the difference between `TableWriteExec` and `V2TableWriteExec` by name.

Reply:
Yes agree the name might be too general. Not sure about `TableWriteExecWithV1Support` though since it sounds like only support v1 write. What about `TableWriteExecHelper` (since it just expose one helper method for writing v1/v2 tables) or `TableWriteExecUtil`? I'm not very good at naming things ...

Reply:
Both work for me.